pkg/controller/storagecluster/reconcile.go

```diff
@@ -744,11 +745,33 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephClus
 		},
 	}
 
-	if len(sc.Spec.MonDataDirHostPath) != 0 {
+	// If both `MonDataDirHostPath` and `monPVCTemplate` is provided, the mons will
+	// be provisioned on the provided `monPVCTemplate` and `monDataDirHostPath`.
+	if len(sc.Spec.MonDataDirHostPath) != 0 && sc.Spec.MonPVCTemplate != nil {
```

Comment:
can we extract `len(sc.Spec.MonDataDirHostPath) != 0` and `sc.Spec.MonPVCTemplate != nil` into vars like `monDataDirHostPathExists` and `monPVCTemplate` instead of evaluating them every time? It'll be easier to read the code as english too.

Reply:
done